woodwork/table_accessor.py

```diff
@@ -458,7 +478,7 @@ def _get_subset_df_with_schema(self, cols_to_include):
         retaining all typing information and maintaining the DataFrame's column order.
         '''
         assert all([col_name in self._schema.columns for col_name in cols_to_include])
-        cols_to_include = [col_name for col_name in self._dataframe.columns if col_name in cols_to_include]
+        cols_to_include = [col_name for col_name in cols_to_include if col_name in self._dataframe.columns]
```

Comment:
Was the reason for switching this so that the order of the columns of a list input to `__getitem__` would be maintained in the new dataframe?

It's causing column order issues with `test_select_ltypes_no_match_and_all`, because, in that case, we want the column order to match the DataFrame's order....

I'm thinking it might be best to go back to the original way you were creating the new DataFrame with Woodwork so that we can use this method for maintaining column order. Or we could add a param for `use_dataframe_order` or something so that we can decide when to follow DataFrame vs input order. 

Sorry for the confusion here!!

Reply:
Yes, that was the reason for the switch. Okay, I'll look into the param option.

Reply:
✅ 3d0e771